src/backend/distributed/sql/udfs/citus_prepare_pg_upgrade/latest.sql

```diff
@@ -67,6 +69,25 @@ BEGIN
        objects.colocationid
     FROM citus.pg_dist_object objects,
          pg_catalog.pg_identify_object_as_address(objects.classid, objects.objid, objects.objsubid) address;
+
+    SELECT count(*) > 0
+    FROM pg_depend
+    WHERE objid::regclass::text = 'citus_stat_activity'
+    AND refobjid IN (
+        SELECT oid FROM pg_extension WHERE extname = 'citus'
+    )
+    INTO citus_depends_on_citus_stat_activity;
+
+    -- drop citus_stat_activity to recreate in citus_prepare_pg_upgrade
+    -- so it has all the columns in pg_stat_activity, even if the upgrade changes them
+    -- also we need to check if it exists to make this function idempotent
+    IF citus_depends_on_citus_stat_activity THEN
+        ALTER EXTENSION citus DROP VIEW citus_stat_activity;
+        ALTER EXTENSION citus DROP TYPE citus_stat_activity_one_node;
+    END IF;
+
+    DROP VIEW IF EXISTS pg_catalog.citus_stat_activity;
+    DROP TYPE IF EXISTS pg_catalog.citus_stat_activity_one_node;
```

Comment:
Instead of dropping the VIEW in prepare_pg_upgrade and recreating it finish_pg_upgrade, I think an easier  way is to create the view using `CREATE OR REPLACE`. Then there's no need to do all this pg_depend stuff (I think).

Reply:
Hmm, I think we tried that didn't work for some reason. But let me retry that actually.

Reply:
I think that was because we also need a type `citus_stat_activity_one_node` which we have to drop (because there is not create_or_replace_type in Postgres. Maybe we can try `worker_create_or_replace_object` hack

Reply:
I removed this but for the record, it was in case a column was dropped from `pg_stat_activity`. If you don't drop the table before upgrade, and a column is dropped in upgrade it failed.